eclipse-che/che-server
# feat: GitHub OAuth authentication with embededOAuthAPI

Signed-off-by: xbaran4 <pbaran@redhat.com>



### What does this PR do?


### Screenshot/screencast of this PR

Enables OAuth flow for GitHub without Keycloak. Using embededOAuthAPI

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20497

### How to test this PR?

1.Deploy quay.io/pbaran/che-server:github10 with devworkspaces.
2. Create GitHub OAuth App in your account. (https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app)
2.1 Application name => Che (But should not matter)
2.2 Homepage URL => $CHE_HOST
2.3 Authorization callback URL => $CHE_HOST/api/oauth/callback
2.4 Register application and generate a new client secret
3. Use your client id and client secret to create k8s github-oauth-config secret:
```shell
kubectl apply -f - <<EOF
kind: Secret
apiVersion: v1
metadata:
  name: github-oauth-config
  namespace: <Che namespace. Default eclipse-che> 
  labels:
    app.kubernetes.io/part-of: che.eclipse.org
    app.kubernetes.io/component: oauth-scm-configuration
  annotations:
    che.eclipse.org/oauth-scm-server: github
type: Opaque
data:
  id: <...> 
  secret: <...> 
EOF
```
(https://www.eclipse.org/che/docs/che-7/administration-guide/configuring-authorization/#configuring-github-oauth_che)
4. Right now you have to restart che-server pod to have the credentials mounted in che.
5. Try to create factory from private repo on your github.

Note: right now you have to have older dashboard version i.e. 7.39.2 because of a bug: https://github.com/eclipse/che/issues/20835

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
